File: Split_the_Array.py
class Solution:
    def isPossibleToSplit(self, nums: List[int]) -> bool:
        count = {}

        for n in nums:
            count[n] = 1 + count.get(n, 0)
        
        for val in count.values():
            if val > 2:
                return False
        
        return True

File: Split_the_Array.cpp
class Solution {
public:
    bool isPossibleToSplit(vector<int>& nums) {
        unordered_map<int, int> count;

        for (int n : nums) {
            count[n]++;
        }
        
        for (auto& pair : count) {
            if (pair.second > 2) {
                return false;
            }
        }
        
        return true;        
    }
};
